shopfloor_reception_shipment_advice/services/reception.py

```diff
@@ -0,0 +1,223 @@
+        )
+        if not pickings:
+            return self._response_for_select_document(
+                message=self.msg_store.shipment_nothing_to_unload(shipment)
+            )
+        self.work.current_shipment_advice = shipment
+        return self._response_for_select_move(picking=pickings)
+
+    def _scan_document__get_handlers_by_type(self):
+        res = super()._scan_document__get_handlers_by_type()
+        res["shipment"] = self._scan_document__by_shipment
+        res["dock"] = self._scan_document__by_dock
+        return res
+
+    def _response_for_select_move_get_data(self, picking):
+        data = {}
+        if picking:
+            data = super()._response_for_select_move_get_data(picking)
+        if self.shipment_advice:
+            data["shipment"] = self.data_detail.shipment_advice_detail(
+                self.shipment_advice
+            )
+            data.pop("picking", None)
+        return data
+
+    def _response_for_manual_selection_shipment(self, dock):
+        shipments = self.env["shipment.advice"].search(
+            self._domain_shipment_advice(dock=dock, today_only=True),
+            order="arrival_date desc",
+        )
+        data = {
+            "dock": self.data.dock(dock),
+            "shipments": [
+                self.data.shipment_advice(shipment) for shipment in shipments
+            ],
+        }
+        return self._response(next_state="manual_selection_shipment", data=data)
+
+    def _select_move_to_work_on(self, moves, message_code="product"):
+        """Select a move with still some quantity to process.
+
+        When working from shipment advice, there can be multiple moves selected for
+        the same barcode.
+        """
+        message = None
+        for move in moves:
+            message = self._check_move_available(move, message_code)
+            if not message:
+                # Found a move that can be worked on
+                return move, message
+        move = fields.first(moves)
+        return move, self._check_move_available(move, message_code)
+
+    def _scan_line_shipment__by_product(self, shipment, product):
+        moves = shipment.planned_move_ids.filtered(lambda m: m.product_id == product)
+        move, message = self._select_move_to_work_on(moves, "product")
+        picking = move.picking_id
+        if message:
+            return self._response_for_select_move(picking, message=message)
+        return self._scan_line__find_or_create_line(picking, move)
+
+    def _scan_line_shipment__by_packaging(self, shipment, packaging):
+        moves = shipment.planned_move_ids.filtered(
+            lambda m: packaging in m.product_id.packaging_ids
+        )
+        move, message = self._select_move_to_work_on(moves, "package")
+        picking = move.picking_id
+        if message:
+            return self._response_for_select_move(picking, message=message)
+        return self._scan_line__find_or_create_line(picking, move)
```

Comment:
```suggestion
    def _scan_line_shipment__by_moves(self, moves):
        move, message = self._select_move_to_work_on(moves, "product")
        picking = move.picking_id
        if message:
            return self._response_for_select_move(picking, message=message)
        return self._scan_line__find_or_create_line(picking, move)
    
    def _scan_line_shipment__by_product(self, shipment, product):
        moves = shipment.planned_move_ids.filtered(lambda m: m.product_id == product)
        return self._scan_line_shipment__by_moves(moves)

    def _scan_line_shipment__by_packaging(self, shipment, packaging):
        moves = shipment.planned_move_ids.filtered(
            lambda m: packaging in m.product_id.packaging_ids
        )
        return self._scan_line_shipment__by_moves(moves)
```

Reply:
But then just call it _scan_line_shipment__by_move and pass the move instead of the moves?